src/components/ModelingMachineProvider.tsx

```diff
@@ -1727,6 +1762,18 @@ export const ModelingMachineProvider = ({
     }
   )
 
+  // Add debug function to window object
+  useEffect(() => {
+    // @ts-ignore - we're intentionally adding this to window
+    window.getModelingState = () => {
+      const modelingState = modelingActor.getSnapshot()
+      return {
+        modelingState,
+        id: modelingState._nodes[modelingState._nodes.length - 1].id,
+      }
+    }
+  }, [modelingActor])
```

Comment:
Hey @Irev-Dev I adjusted this to use `getSnapshot` instead so the listener only has to get applied once, sorry editing it deleted your comment.